source-control-sample does not compile - TS2345: Argument of type 'readonly WorkspaceFolder[]' is not assignable to parameter of type 'any[]'

Compiling samples fails because of error in `src/extension.ts` file from **source-control-sample** project.

```
> source-control-sample@0.0.1 compile
> tsc -p ./

src/extension.ts:251:74 - error TS2345: Argument of type 'readonly WorkspaceFolder[]' is not assignable to parameter of type 'any[]'.
  The type 'readonly WorkspaceFolder[]' is 'readonly' and cannot be assigned to the mutable type 'any[]'.

251   workspaceFolderIndex = vscode.workspace.workspaceFolders && firstIndex(vscode.workspace.workspaceFolders, (folder1: any) => folder1.uri.toString() === workspaceFolderUri!.toString());
                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/extension.ts:251

node:child_process:946
    throw err;
    ^

Error: Command failed: npm run compile
    at checkExecSyncError (node:child_process:871:11)
    at Object.execSync (node:child_process:943:15)
    at tryRun (d:\dev\github\NetOfficeFw\extras\vscode-extension-samples\.scripts\run-script.js:18:19)
    at Object.<anonymous> (d:\dev\github\NetOfficeFw\extras\vscode-extension-samples\.scripts\run-script.js:27:3)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
  status: 2,
  signal: null,
  output: [ null, null, null ],
  pid: 17244,
  stdout: null,
  stderr: null
}

Node.js v18.12.1
```

I'm sorry I don't understand all of the computer codes and stuff. Have u any pictures to screenshot and save me by any chance? I don't have the software to extract anything
 I only have a phone

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Jozef Izso ***@***.***>
Sent: Thursday, December 29, 2022 7:33:21 PM
To: microsoft/vscode-extension-samples ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [microsoft/vscode-extension-samples] source-control-sample does not compile - TS2345: Argument of type 'readonly WorkspaceFolder[]' is not assignable to parameter of type 'any[]' (Issue #795)


Compiling samples fails because of error in src/extension.ts file from source-control-sample project.

> ***@***.*** compile
> tsc -p ./

src/extension.ts:251:74 - error TS2345: Argument of type 'readonly WorkspaceFolder[]' is not assignable to parameter of type 'any[]'.
  The type 'readonly WorkspaceFolder[]' is 'readonly' and cannot be assigned to the mutable type 'any[]'.

251   workspaceFolderIndex = vscode.workspace.workspaceFolders && firstIndex(vscode.workspace.workspaceFolders, (folder1: any) => folder1.uri.toString() === workspaceFolderUri!.toString());
                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/extension.ts:251

node:child_process:946
    throw err;
    ^

Error: Command failed: npm run compile
    at checkExecSyncError (node:child_process:871:11)
    at Object.execSync (node:child_process:943:15)
    at tryRun (d:\dev\github\NetOfficeFw\extras\vscode-extension-samples\.scripts\run-script.js:18:19)
    at Object.<anonymous> (d:\dev\github\NetOfficeFw\extras\vscode-extension-samples\.scripts\run-script.js:27:3)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
  status: 2,
  signal: null,
  output: [ null, null, null ],
  pid: 17244,
  stdout: null,
  stderr: null
}

Node.js v18.12.1


—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/vscode-extension-samples/issues/795>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A42JPMTWPRQZHNRY244WKW3WPXRQDANCNFSM6AAAAAATMK36UE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

--- a/source-control-sample/src/util.ts
+++ b/source-control-sample/src/util.ts
@@ -1,4 +1,4 @@
-export function firstIndex<T>(array: T[], fn: (t: T) => boolean): number {
+export function firstIndex<T>(array: readonly T[], fn: (t: T) => boolean): number {
 	for (let i = 0; i < array.length; i++) {
 		if (fn(array[i])) {
 			return i;

